VirtPub/Services/GameService.cs

```diff
@@ -0,0 +1,180 @@
+            var response = await Client.GetAsync(
+                "api/v1.0/Table");
+
+            using var responseStream = await response.Content.ReadAsStreamAsync();
+
+            return await JsonSerializer.DeserializeAsync
+                <IEnumerable<TableModel>>(responseStream);
+        }
+
+        public async Task<string> CreateTable(GameLinksModel Game)
+        {
+            var table = new TableModel();
+            table.id = Guid.NewGuid();
+            table.isPrivate = false;
+            table.game = Game;
+
+            var json = Newtonsoft.Json.JsonConvert.SerializeObject(table);
+            var data = new StringContent(json, UnicodeEncoding.UTF8, "application/json");
+
+            var response = await Client.PostAsync(
+                "api/v1.0/Table", data);
+
+            string result = response.Content.ReadAsStringAsync().Result;
+            return result;
+        }
+
+        public void AddUserToUserList(string userName, string group, string connectionId)
+        {
+            try
+            {
+                var isUserAdmin = WillUserBeTableAdmin(group);
+
+                var newUser = users.Where(z => z.UserName == userName).ToList();
+
+                if (newUser.Count == 0)
+                {
+
+                    users.Add(new ConnectedUser()
+                    {
+                        UserName = userName,
+                        ConnectionId = connectionId,
+                        Group = group,
+                        IsAdmin = isUserAdmin
+                    });
+                }
+            }
+            catch (System.Exception)
+            {
+                System.Console.WriteLine("Unable to add the user to the list: Users");
+            }
+        }
+
+        private bool WillUserBeTableAdmin(string group)
+        {
+            if (users.Where(x => x.Group == group && x.IsAdmin == true).Count() == 0)
+                return true;
+
+            return false;
+        }
+
+        public async Task RemoveUserFromUserList(string userName)
+        {
+            try
+            {
+                var userToRemove = users.Where(x => x.UserName == userName).First();
+               users.Remove(userToRemove);
+
+                if (userToRemove != null && userToRemove.IsAdmin && users.Count() > 0)
+                {
+                    users[0].IsAdmin = true;
+                }
+            }
+            catch (System.Exception)
+            {
+                System.Console.WriteLine("Unable to remove the user from the list: Users");
+            }
+        }
+
+        public async Task<List<ScoreboardModel>> GetScoreboardByGameId(string id) 
+        {
+            var response = await Client.GetAsync(
+              $"api/v1.0/ScoreBoard/{id}");
+
+            using var responseStream = await response.Content.ReadAsStreamAsync();
+
+            return await JsonSerializer.DeserializeAsync
+                <List<ScoreboardModel>>(responseStream);
+        }
```

Comment:
Det vi tänkte på här är "separations of concerns", Det hade kunnat blivit snyggare / lättare att förstå ifall detta var indelat i olika services. T.ex. en för Tables, en för Scoreboard osv. Eftersom det inte är så många calls just nu gör det inte så mycket, men ifall man ska skala upp applikationen kan det bli rörigt.

Reply:
Något vi pratat om och kommer ta i beaktning. 👍🏻